sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -363,6 +363,69 @@ trait Row extends Serializable {
     false
   }
 
+  protected def canEqual(other: Any) = other.isInstanceOf[Row]
```

Comment:
`other` could be `null`. And `=(other.getClass == this.getClass)`, so we don't need to override it within its subclasses.
BTW: Should we mark it identical for a GenericRow and a MutableRow if they contains exactly the same elements?